tests/renderer.test.ts

```diff
@@ -126,4 +275,142 @@ test('can remove unused guides', () => {
       }
     },
   ]);
+});
+
+it('can take a custom setup', () => {
+  const document: DocumentInterface = {
+    createElement(tagName: string) {
+      return { tagName, className: '', style: {} };
+    }
+  };
+
+  const body: ContainerInterface = {
+    children: [],
+
+    appendChild(child: ElementInterface) {
+      this.children.push(child);
+    },
+
+    removeChild(child: ElementInterface) {
+      this.children = this.children.filter((existing: ElementInterface) => existing !== child);
+    }
+  };
+
+  const renderer = new Renderer({
+    document,
+    container: body,
+    setup: ({ element, edge }) => element.className = `my-${edge.type}-${edge.direction}`
+  });
+
+  renderer.draw([
+    new Edge({ direction: 'horizontal', position: 100, type: 'page' }),
+    new Edge({ direction: 'vertical', position: 300, type: 'normal' }),
+  ]);
+
+  expect(body.children).toEqual([
+    {
+      tagName: 'div',
+      className: 'my-page-horizontal',
+      style: { top: '100px', left: null, }
+    },
+    {
+      tagName: 'div',
+      className: 'my-normal-vertical',
+      style: { left: '300px', top: null, }
+    }
+  ]);
+});
+
+it('can take a custom positioner', () => {
+  const document: DocumentInterface = {
+    createElement(tagName: string) {
+      return { tagName, className: '', style: {} };
+    }
+  };
+
+  const body: ContainerInterface = {
+    children: [],
+
+    appendChild(child: ElementInterface) {
+      this.children.push(child);
+    },
+
+    removeChild(child: ElementInterface) {
+      this.children = this.children.filter((existing: ElementInterface) => existing !== child);
+    }
+  };
+
+  const renderer = new Renderer({
+    document,
+    container: body,
```

Comment:
Thinking about these, we could rename out local const to `container` so that we can use the object shorthand.